Compile-time error trying to build code and run tests

# Summary

I'm seeing a compile-time error when I try to build and run the tests. Any insight or tip about what I'm doing wrong is appreciated!

# Version
This is with oneMKL pulled from github, git commit `596ba0a7ce75547698a311f607629bbd4fac03ac`

# Environment

* HW you use: Iris Gen9
* Backend library version: OpenCL
* OS name and version: linux, openSUSE, 15.2
* Compiler version:  
```
> dpcpp -v
Intel(R) oneAPI DPC++ Compiler 2021.1 (2020.10.0.1113)
```

# Steps to reproduce

```
    blas_root=$PWD
    # first get lapack                                                                                                                                                                                                              
    git clone https://github.com/Reference-LAPACK/lapack.git
    cd lapack/
    mkdir build
    cd build/
    cmake -DBUILD_SHARED_LIBS=ON -DCBLAS=ON -DCMAKE_INSTALL_LIBDIR=${blas_root} -DCMAKE_INSTALL_PREFIX=${blas_root} ..
    cmake --build . -j4 --target install
    cd ../..

    git clone https://github.com/oneapi-src/oneMKL.git
    cd oneMKL
    rm -rf build
    mkdir build
    cd build
    CXX=`which dpcpp` cmake ../ -DMKL_ROOT=$MKLROOT -DREF_BLAS_ROOT=${blas_root} -DSYCL_LIBRARY=${SDK_ROOT}/compiler/latest/linux/lib/libsycl.so
    cmake --build . -j4
    ctest
```
# Observed behavior
The code isn't compiling for me. Am I doing something obviously wrong? The build output ends with:
```
...
Scanning dependencies of target gtest_main
[ 96%] Building CXX object deps/googletest/CMakeFiles/gtest_main.dir/src/gtest_main.cc.o
[ 96%] Building CXX object tests/unit_tests/blas/level2/CMakeFiles/blas_level2_rt.dir/spr_usm.cpp.o
[ 96%] Linking CXX shared library ../../lib/libgtest_main.so
[ 96%] Built target gtest_main
[ 96%] Building CXX object tests/unit_tests/blas/level2/CMakeFiles/blas_level2_rt.dir/spr2_usm.cpp.o
Scanning dependencies of target test_main_rng_ct
[ 96%] Building CXX object tests/unit_tests/CMakeFiles/test_main_rng_ct.dir/main_test.cpp.o
Scanning dependencies of target test_main_blas_ct
[ 96%] Building CXX object tests/unit_tests/CMakeFiles/test_main_blas_ct.dir/main_test.cpp.o
Scanning dependencies of target test_main_rng_rt
[ 96%] Building CXX object tests/unit_tests/CMakeFiles/test_main_rng_rt.dir/main_test.cpp.o
[ 97%] Building CXX object tests/unit_tests/blas/level2/CMakeFiles/blas_level2_rt.dir/spmv_usm.cpp.o
[ 98%] Linking CXX executable ../../bin/test_main_rng_ct

[ FATAL ] /home/bertoni/gpu_tests/testing/source/benchmarks/conformance/r.oneMKL/oneMKL/deps/googletest/include/gtest/internal/gtest-param-util.h:562:: Condition test_param_names.count(param_name) == 0 failed. Duplicate paramet\
erized test name 'Intel_R__Xeon_R__CPU_E3_1585_v5___3_50GHz', in /home/bertoni/gpu_tests/testing/source/benchmarks/conformance/r.oneMKL/oneMKL/tests/unit_tests/rng/statistics_check/uniform.cpp line 98

CMake Error at /soft/packaging/spack-builds/linux-rhel7-x86_64/gcc-9.3.0/cmake-3.18.2-mwdhwbhynfd7dcpegq4iq6xkzvavcmsh/share/cmake-3.18/Modules/GoogleTestAddTests.cmake:77 (message):
  Error running test executable.

    Path: '/home/bertoni/gpu_tests/testing/source/benchmarks/conformance/r.oneMKL/oneMKL/build/bin/test_main_rng_ct'
    Result: Child aborted
    Output:


Call Stack (most recent call first):
  /soft/packaging/spack-builds/linux-rhel7-x86_64/gcc-9.3.0/cmake-3.18.2-mwdhwbhynfd7dcpegq4iq6xkzvavcmsh/share/cmake-3.18/Modules/GoogleTestAddTests.cmake:173 (gtest_discover_tests_impl)


gmake[2]: *** [tests/unit_tests/CMakeFiles/test_main_rng_ct.dir/build.make:133: bin/test_main_rng_ct] Error 1
gmake[2]: *** Deleting file 'bin/test_main_rng_ct'
gmake[1]: *** [CMakeFiles/Makefile2:922: tests/unit_tests/CMakeFiles/test_main_rng_ct.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....

```

# Expected behavior
I expect it to compile and run.

--- a/tests/unit_tests/main_test.cpp
+++ b/tests/unit_tests/main_test.cpp
@@ -121,8 +121,13 @@ int main(int argc, char** argv) {
 #if defined(ENABLE_MKLCPU_BACKEND) || defined(ENABLE_NETLIB_BACKEND)
     local_devices.push_back(cl::sycl::device(cl::sycl::host_selector()));
 #endif
-    for (int i = 0; i < local_devices.size(); i++) {
-        devices.push_back(&(local_devices[i]));
+#define GET_NAME(d) (d).template get_info<cl::sycl::info::device::name>()
+    for (auto& local_dev : local_devices) {
+        // Test only unique devices
+        if (std::find_if(devices.begin(), devices.end(), [&](cl::sycl::device* dev) {
+                return GET_NAME(*dev) == GET_NAME(local_dev);
+            }) == devices.end())
+            devices.push_back(&local_dev);
     }
 
     // start Google Test pickup and output

